### ISSUE TITLE
source code

### ISSUE THREAD

**Issue by litao1381** [steffensbola/a6300_multi_exposure_dataset#1]
Can you provide source code?

**Comment by steffensbola** [comment]
Hi,
Thanks for getting in touch. Unfortunately, I have managed to somehow mislay the code used to train the network in the original paper. It should not be too hard to reproduce, but I haven't had the time/motivation to do it.

We have two new and improved models which are available at:
[U-CAN_32](https://github.com/steffensbola/u_can_32_image_exposure)
[ReExposeNet](https://github.com/lrvm98/ReExposeNet)

You should be able to use it on any miss exposed image without having to retrain. We find it to be very robust. It introduces some enhancements over the paper we have published in 2018, is able to work on distinct image resolutions, and yields less color aberration.

The most important differences from these new models to the old one are:
- We got rid of the adversarial network: we found it to result in aberrations and longer training time. An evaluation with several image quality metrics has shown that a model with pixel-wise and structural loss yields better results;
- We modified the convolutional block so it covers a broader feature space. This came useful for inference in high-resolution images (>= Full-HD);
- We modified the output activation function to allow for pixel values greater or smaller than the representation range. This came useful in model training, once it allows better response in critical parts of the image.
- We implement a custom, content-based loss function, which puts more emphasis on dark and bright blocks of the image. These are more prone to saturation.

These models were trained with Keras / TensorFlow 1.11.0 on a Titan X GPU. The only limitation is that your input image is in the RGB format, with a resolution multiple of 8 (e. g. 512x512, 1080x720, 1920x1080,...).  The aspect ratio does not matter.

**Comment by steffensbola** [comment]
[](https://github.com/steffensbola/u_can_32_image_exposure)

**Comment by litao1381** [comment]
thank you
